File: examples/js/03_node_client_api/students_test.ts
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

import { touca } from '@touca/node';
import { parse_profile } from './students';

(async () => {
  await touca.configure();
  for (const username of ['alice', 'bob', 'charlie']) {
    touca.declare_testcase(username);

    const student = await touca.scoped_timer('parse_profile', async () =>
      parse_profile(username)
    );
    touca.add_assertion('username', student.username);
    touca.add_result('fullname', student.fullname);
    touca.add_result('birth_date', student.dob);
    touca.add_result('gpa', student.gpa);
    touca.add_metric('external_source', 150);

    await touca.post();
    await touca.save_json(`touca_${username}.json`);
    await touca.save_binary(`touca_${username}.binary`);
    touca.forget_testcase(username);
  }
  await touca.seal();
})();

File: examples/java/03_java_core_api/src/main/java/io/touca/examples/core/StudentsTest.java
// Copyright 2022 Touca, Inc. Subject to Apache-2.0 License.

package io.touca.examples.core;

import io.touca.Touca;
import java.io.IOException;

public final class StudentsTest {
  public static void main(String[] args) throws IOException {
    Touca.configure(options -> {
      options.offline = true;
    });
    if (!Touca.isConfigured()) {
      System.err.print(Touca.configurationError());
      System.exit(1);
    }
    for (String username : new String[] { "alice", "bob", "charlie" }) {
      Touca.declareTestcase(username);
      Touca.startTimer("find_student");
      Student student = Students.findStudent(username);
      Touca.stopTimer("find_student");

      Touca.addTypeAdapter(Course.class, course -> {
        return course.name;
      });
      Touca.assume("username", student.username);
      Touca.check("fullname", student.fullname);
      Touca.check("birth_date", student.dob);

      Touca.scopedTimer("calculate_gpa", () -> {
        Touca.check("gpa", Students.calculateGPA(student.courses));
      });
      Touca.addMetric("external_source", 1500);

      Touca.post();
      Touca.saveJson(String.format("touca_%s.json", username), null);
      Touca.saveBinary(String.format("touca_%s.bin", username), null);
      Touca.forgetTestcase(username);
    }
    Touca.seal();
  }
}
